core/src/net/udp_state.rs

```diff
@@ -44,31 +53,41 @@ impl UdpState {
         let mut sent_bytes: usize = 0;
         let mut interrupts = 0;
         while let Some((addr, mut frame)) = self.outbound_queue.pop_front() {
-            frame.make_contiguous();
-            match self.socket.send_to(frame.bytes(), addr) {
-                Ok(n) => {
-                    // This really shouldn't happen, and can lead to inconsistent network messages
-                    assert_eq!(n, frame.len(), "A UDP frame was written incompletely!");
-                    sent_bytes += n;
-                }
-                Err(ref err) if would_block(err) => {
-                    // re-insert the data at the front of the buffer and return
-                    self.outbound_queue.push_front((addr, frame));
-                    return Ok(sent_bytes);
-                }
-                Err(err) if interrupted(&err) => {
-                    // re-insert the data at the front of the buffer
-                    self.outbound_queue.push_front((addr, frame));
-                    interrupts += 1;
-                    if interrupts >= MAX_INTERRUPTS {
+            if frame.len() > self.max_packet_size {
```

Comment:
Hmm, is this right? Couldn't it happen that some actor has a different chunk size and already serialised it in a contiguous manner? Shouldn't it be the make_contiguous call below that fails if we don't give it a large enough buffer or something?

Reply:
The BufferConfig which (maybe) determines max_packet_size is the one used for the NetworkThread, so I'm assuming the remote systems NetworkConfig is set-up with the same BufferConfig. The line imposes that we won't send packages larger than we can receive. I could remove that check entirely as the udp send or receive will trigger the error/warning instead if the frame is too big.

Reply:
Yeah, I don't think we should make this assumption. Let it send whatever it can, and let it fail on the receiver side if necessary. I think we should assume that Kompact systems in general won't share the same configuration.

Reply:
But you do need to make sure that non-contiguous data can be made contiguous on the sending side. I think that's the only check we should have, since we definitely should not send broken packages!